source/renderer/app/stores/HardwareWalletsStore.ts

```diff
@@ -1194,7 +1243,33 @@ export default class HardwareWalletsStore extends Store {
         ) || {};
       // @ts-ignore ts-migrate(2339) FIXME: Property 'path' does not exist on type '{}'.
       devicePath = activeDevice.path || path || newConnectionData.path || null;
-      await this._getExtendedPublicKey(devicePath, walletId, address);
+
+      const extendedPublicKey = await this._requestExtendedPublicKey(
+        devicePath,
+        walletId,
+        address
+      );
+      const recognizedSoftwareWallet = await this._recognizeSoftwareWalletByExtendedPublicKey(
+        { extendedPublicKey }
+      );
+
+      if (recognizedSoftwareWallet) {
+        await this._handleIdentifiedSoftwareWallet({
+          recognizedSoftwareWallet,
+          extendedPublicKey,
+          forcedPath: devicePath,
+          walletId,
+          address,
+        });
```

Comment:
@szymonmaslowski following to the above comment. Now you have 2 functions that are connected to identifying / recognizing. It is better to say "ok, now get me an extended public key and then recognize"

Reply:
Sorry I didn't get your comment. By two functions do you mean the `_handleIdentifiedSoftwareWallet` and `_identifyAndHandleSoftwareWallet`? The `_handleIdentifiedSoftwareWallet` method is responsible only for the case when the wallet was found. On the other hand the `_identifyAndHandleSoftwareWallet` method encapsulates all steps including the extended public key request, finding the device and handling it accordingly, and it is there for a reduction of code duplication, but I believe with further refactoring it may be broken to few less generic code blocks and distributed to the proper places in the logic (for example part regarding the transaction to the transaction logic etc.).